dali/pipeline/operators/operator.h

```diff
@@ -212,6 +232,101 @@ class Operator : public OperatorBase {
   template <typename B = Backend>
   typename std::enable_if<!std::is_same<B, GPUBackend>::value>::type
   SyncHelper(int /*unused*/, Workspace<B> */*unused*/) {}
+
+  template <typename B = Backend>
+  typename std::enable_if<std::is_same<B, GPUBackend>::value>::type
+  Flatten(Workspace<Backend> *ws) {
+    seq_sizes_.resize(input_sets_, 1);
+    for (int i = 0; i < input_sets_; ++i) {
+      auto &input = ws->template MutableInput<Backend>(i);
+
+      const std::vector<Dims> old_shapes = input.shape();
+      // only if Tensors are 4-d
+      if (old_shapes[0].size() == 4) {
+        // size of seq is the first dim in each tensor
+        seq_sizes_[i] = old_shapes[0][0];
+
+        std::vector<Dims> new_shapes;
+        for (const Dims &old_shape : old_shapes) {
+          // getting only the need 3 dimensions
+          Dims new_shape(old_shape.begin() + 1, old_shape.end());
+          for (int s = 0; s < seq_sizes_[i]; ++s) {
+            new_shapes.emplace_back(new_shape);
+          }
+        }
+        input.Resize(new_shapes);
+        input.SetLayout(DALI_NHWC); 
+      }
+    }
+  }
+
+  template <typename B = Backend>
+  typename std::enable_if<!std::is_same<B, GPUBackend>::value>::type
+  Flatten(Workspace<B> */*unused*/) {}
+
+  template <typename B = Backend>
+  typename std::enable_if<std::is_same<B, GPUBackend>::value>::type
+  Unflatten(Workspace<Backend> *ws) {
+    // TODO(spanev): Handle ops where OpSchema::NumInput != OpSchema::NumOuput?
+    for (int idx = 0; idx < input_sets_; ++idx) {
+      if (seq_sizes_[idx] > 1) {
+        auto &input = ws->template MutableInput<Backend>(idx);
+        auto &output = ws->template Output<Backend>(idx);
+        const std::vector<Dims>& old_shapes_input = input.shape();
+        const std::vector<Dims>& old_shapes_output = output.shape();
+        std::vector<Dims> new_shapes_input;
+        std::vector<Dims> new_shapes_output;
+        std::cout << "old in shapes size " << old_shapes_input.size() << std::endl;
+        std::cout << "old out shapes size " << old_shapes_output.size() << std::endl;
+        for (unsigned int i = 0; i < old_shapes_input.size(); i += seq_sizes_[idx]) {
+          Dims shape_input(old_shapes_input[i]);
```

Comment:
More efficient:
```suggestion
          Dims shape_input;
          shape_input.reserve(old_shapes_input[i].size());
          shape_input.push_back(seq_sizes_[idx]);
          shape_input.insert(shape_input.end(), old_shapes_input[i].begin(), old_shapes_input[i].end());
          new_shapes_input.push_back(std::move(shape_input));
```

Reply:
```
new_shapes_input.push_back(std::move(shape_input));
```
wouldn't be produce the same assembly as
```
new_shapes_input.emplace_back(shape_input);
```
??